### ISSUE TITLE
pulumi 2.13.0 introduces issue with eks package

### ISSUE THREAD

**Issue by jaxxstorm** [pulumi/pulumi#5705]
Since pulumi 2.13.0, we're having reports from customers that the https://github.com/pulumi/pulumi-eks package no longer works correctly. An example error:

```
Error: Running program 'foo' failed with an unhandled exception:
    Error: Unable to deserialize resource urn:pulumi:dev::foo::aws:iam/role:Role::foo, no module is registered for iam/role.
```

this was reported here: https://github.com/pulumi/examples/issues/826

and it looks like this might possibly be related: https://github.com/pulumi/pulumi/pull/5645

**Comment by jaxxstorm** [comment]
the current recommended action is to downgrade to pulumi 2.12.1

If you've installed using homebrew, you can do this:

```
curl -fsSL https://get.pulumi.com | sh -- version v2.12.1
```

You'll need to make sure this version is first in your `PATH` - it gets installed to `~/.bin/pulumi`

**Comment by justinvp** [comment]
Summary of initial investigation:

https://github.com/pulumi/pulumi/pull/5041/ added support for serializing resource references. This shipped with 2.13.0 and 2.13.1 of the CLI / language SDKs.

Previously, when serializing a resource, we would simply serialize it’s ID.

Now, when serializing a resource, we check the monitor to see if it supports "resourceReferences" (which CLI 2.13.0 and 2.13.1 reports it does), in which case, instead of serializing the ID, we serialize a special `specialResourceSig` object with the URN and ID.

https://github.com/pulumi/pulumi/blob/ff226499259c7b304c87adc7665eda536d47011a/sdk/nodejs/runtime/rpc.ts#L332-L351

During deserialization, if we see a `specialResourceSig` we attempt to instantiate the resource object. However, there's still some remaining work to do to fully support this. In particular, each generated package needs to be updated to register all the resources that can be instantiated.

It looks like EKS is running into a case where some resources are being serialized, but since the 2.13.0/2.13.1 CLI is reporting "resourceReferences" as being a supported feature, the resources are being serialized in this new way (instead of just the ID) and then during deserialization the resource can’t be reinstantiated because we haven’t registered any resources yet.

To quickly address this issue, we can update the CLI to report that "resourceReferences" isn't supported yet, until we've completed the work to be able to instantiate the resources.
